jetty-core/jetty-fcgi/jetty-fcgi-server/src/test/java/org/eclipse/jetty/fcgi/server/AbstractHttpClientServerTest.java

```diff
@@ -65,33 +61,46 @@ public void start(Handler handler) throws Exception
         QueuedThreadPool clientThreads = new QueuedThreadPool();
         clientThreads.setName("client");
         clientConnector.setExecutor(clientThreads);
-        // TODO: restore leak tracking.
         if (clientBufferPool == null)
-            clientBufferPool = new ArrayByteBufferPool();
+            clientBufferPool = new ArrayByteBufferPool.Tracking();
         clientConnector.setByteBufferPool(clientBufferPool);
         HttpClientTransport transport = new HttpClientTransportOverFCGI(clientConnector, "");
-        transport.setConnectionPoolFactory(destination -> new LeakTrackingConnectionPool(destination, client.getMaxConnectionsPerDestination())
-        {
-            @Override
-            protected void leaked(LeakDetector<Connection>.LeakInfo leakInfo)
-            {
-                connectionLeaks.incrementAndGet();
-            }
-        });
         client = new HttpClient(transport);
         client.start();
     }
 
     @AfterEach
-    public void dispose() throws Exception
+    public void dispose()
     {
-        System.gc();
-
-        assertThat("Connection Leaks", connectionLeaks.get(), Matchers.is(0L));
-
-        if (client != null)
-            client.stop();
-        if (server != null)
-            server.stop();
+        try
+        {
+            if (serverBufferPool != null)
+            {
+                try
+                {
+                    Awaitility.await().atMost(3, TimeUnit.SECONDS).until(() -> serverBufferPool.getLeaks().size(), Matchers.is(0));
+                }
+                catch (Exception e)
+                {
+                    fail(e.getMessage() + "\n---\nServer Leaks: " + serverBufferPool.dumpLeaks() + "---\n");
+                }
+            }
```

Comment:
Make an `assertLeaks()` method to avoid code duplication.

Reply:
Done.